lib/main.ts

```diff
@@ -1,87 +1,117 @@
-  const entryPoint = sourceFiles.filter(sf => sf.fileName === normalizedFileName)[0];
-  const ms = moduleSymbol(entryPoint);
+
+  const ms = (<any>sourceFile).symbol;
   const rawSymbols = ms ? (typeChecker.getExportsOfModule(ms) || []) : [];
-  return rawSymbols.map(
-      s => { return (s.flags & ts.SymbolFlags.Alias) ? typeChecker.getAliasedSymbol(s) : s; });
+  return rawSymbols.map(s => {
+    if (s.flags & ts.SymbolFlags.Alias) {
+      const resolvedSymbol = typeChecker.getAliasedSymbol(s);
+
+      // This will happen, e.g. for symbols re-exported from external modules.
+      if (!resolvedSymbol.valueDeclaration && !resolvedSymbol.declarations) {
+        return s;
+      }
+      if (resolvedSymbol.name !== s.name) {
+        throw new Error(`Symbol "${resolvedSymbol.name}" was aliased as "${s.name}"`);
+      }
+
+      return resolvedSymbol;
+    } else {
+      return s;
+    }
+  });
 }
 
 function moduleSymbol(sourceFile: ts.SourceFile): ts.Symbol {
   return (<any>sourceFile).symbol;
 }
 
-function getCompilerHost(): ts.CompilerHost {
-  var defaultLibFileName = ts.getDefaultLibFileName({});
-  defaultLibFileName = normalizeSlashes(defaultLibFileName);
-  const host: ts.CompilerHost = {
-    getSourceFile: (sourceName, languageVersion) => {
-      var sourcePath = sourceName;
-      if (sourceName === defaultLibFileName) {
-        sourcePath = ts.getDefaultLibFilePath({});
-      }
-      if (!fs.existsSync(sourcePath)) return undefined;
-      let contents = fs.readFileSync(sourcePath, 'utf-8');
-      return ts.createSourceFile(sourceName, contents, languageVersion, true);
-    },
-    writeFile(name, text, writeByteOrderMark) { fs.writeFile(name, text); },
-    fileExists: (filename) => fs.existsSync(filename),
-    readFile: (filename) => fs.readFileSync(filename, 'utf-8'),
-    getDefaultLibFileName: () => defaultLibFileName,
-    useCaseSensitiveFileNames: () => true,
-    getCanonicalFileName: (filename) => filename,
-    getCurrentDirectory: () => '',
-    getNewLine: () => '\n',
-  };
-  host.resolveModuleNames = getModuleResolver(host);
-  return host;
+function symbolCompareFunction(a: ts.Symbol, b: ts.Symbol) {
+  return a.name.localeCompare(b.name);
 }
 
-function getModuleResolver(compilerHost: ts.CompilerHost) {
-  return (moduleNames: string[], containingFile: string): ts.ResolvedModule[] => {
-    let res: ts.ResolvedModule[] = [];
-    for (let mod of moduleNames) {
-      let lookupRes = ts.nodeModuleNameResolver(mod, containingFile, compilerHost);
-      if (lookupRes.resolvedModule) {
-        res.push(lookupRes.resolvedModule);
-        continue;
-      }
-      lookupRes = ts.classicNameResolver(mod, containingFile, {}, compilerHost);
-      if (lookupRes.resolvedModule) {
-        res.push(lookupRes.resolvedModule);
-        continue;
+/**
+ * Traverses the node tree to construct the text without comments.
+ */
+function getTextWithoutComments(node: ts.Node): string {
+  // TODO: test more
```

Comment:
this is not very actionable todo. either list what exactly needs to be tested or remove.